src/Sylius/Bundle/CoreBundle/EventListener/DefaultUsernameORMListener.php

```diff
@@ -42,13 +43,20 @@ public function onFlush(OnFlushEventArgs $onFlushEventArgs)
      */
     private function processEntities($entities, EntityManagerInterface $entityManager, UnitOfWork $unitOfWork)
     {
-        foreach ($entities as $customer) {
-            if (!$customer instanceof CustomerInterface) {
-                continue;
-            }
-
-            $user = $customer->getUser();
-            if (null === $user) {
+        foreach ($entities as $entity) {
+            if ($entity instanceof CustomerInterface) {
+                $customer = $entity;
+                $user = $customer->getUser();
+                if (null === $user) {
+                    continue;
+                }
+            } elseif ($entity instanceof ShopUserInterface) {
```

Comment:
I'm not sure about this solution. If we want to keep consistency between username and email, we should prevent from changing it. If we allow for this change, then we will receive a bug report that when I change shop users username, it doesn't change... 

Also, we should avoid `elseif` construction. If we want it in core, it should be refactored and extracted to the private method.